src/learner.cc

```diff
@@ -872,6 +917,11 @@ class LearnerIO : public LearnerConfiguration {
 )doc";
       int64_t sz {-1};
       CHECK_EQ(fp.Read(&sz, sizeof(sz)), sizeof(sz));
+
+      if (DMLC_IO_NO_ENDIAN_SWAP == 0){
+        sz = __builtin_bswap64(sz);
```

Comment:
We should avoid using compiler intrinsics like this, since we'd like to support Visual Studio as well. I'm looking into using [`dmlc::ByteSwap()`](https://github.com/dmlc/dmlc-core/blob/master/include/dmlc/endian.h#L51)

Reply:
See my commit.